pkg/serverless/daemon/daemon.go

```diff
@@ -345,7 +342,8 @@ func (d *Daemon) WaitForDaemon() {
 // ComputeGlobalTags extracts tags from the ARN, merges them with any user-defined tags and adds them to traces, logs and metrics
 func (d *Daemon) ComputeGlobalTags(configTags []string) {
 	if len(d.ExtraTags.Tags) == 0 {
-		tagMap := tags.BuildTagMap(d.ExecutionContext.ARN, configTags)
+		ecs := d.ExecutionContext.GetCurrentState()
+		tagMap := tags.BuildTagMap(ecs.ARN, configTags)
```

Comment:
I think we need a `getARN` protected by a lock here because of the following scenario:

1 - ecs := d.ExecutionContext.GetCurrentState() is executed, lock has been acquired and released
2 - another go routine is triggered, modifying ARN
3 - tagMap := tags.BuildTagMap(ecs.ARN, configTags) is executed, ecs.ARN might be in a modification state

Reply:
I'm not sure I follow. Once we call `GetCurrentState()` we have a snapshot of the execution context (`ecs`). If the underlying execution context is modified, the snapshot will be out of date, but there won't be any contention between this goroutine and any other. Does that make sense?

Reply:
In other words, `ecs` is a copy, and only this goroutine has a reference to it.

Reply:
yes you're right, ARN is not written again here